### PR TITLE
i18n improvements and French localization

### PR DESCRIPTION
**Pull request by dannon**
# S'il vous plaît aider si vous parlez français

A first attempt at expanding our localization usage, and implementing french language!

Coverage is spotty, but this is something we can work on.

Masthead:
image

Example of where we need more coverage:
image

### CODE REVIEW

**Comment by dannon** [comment]
Now with masthead support!  (moved screenshot to the top)

**Comment by martenson** [comment]
👏

**Comment by yvanlebras** [comment]
👏

**Comment by abretaud** [comment]
C'est formidable !

**Comment by lecorguille** [comment]
Sure, I can help!
But just one question, how many language do you plan to support?
Is it just for our famous bad skills in English? :P 

Because as far as I know, I never met any French user stuck because they don't understand the user interface (They are because their bam are not sorted :/)

**Comment by dannon** [comment]
@lecorguille Right now we have Japanese, Chinese, and (as of this PR) French.  I don't think we have a limit to the number of languages we want to support other than realizing that the core team certainly can't maintain (or even write them originally!) them all.

What we *can* do is implement a localization framework, and allow native speakers (as @bebatut fixed up the FR version a bit) do the translation as they are willing.

More concretely, the 'framework-side' work is the same for *one* language as it is for *all languages*.  We just need the file ```client/galaxy/scripts/nls/fr/locale.js``` for each locale in question, which we can solicit help from the community to build/maintain.  At *worst*, missing translations will fail through to English.

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: J'aurais gardé `Visualisation` ou hybride : `Visualisation des données`

**Comment by lecorguille** [comment]
@dannon As long as you don't ask us to translate the different tool help section, I'm in!

**Comment by dannon** [comment]
Sounds good!  If you want to fork https://github.com/dannon/galaxy/tree/i18n and PR any changes, feel free!

Next step for finishing off the translation is to change all the 'false' lines to contain the correct french version of the string.  (and uncomment the commented ones, after providing the translation, of course)

**Comment by lecorguille** [comment]
@bebatut To not compete on that, are you currently doing it?

**Comment by bebatut** [comment]
@lecorguille No. I am fighting with conda, g++ and openmpi :)

**Comment by lecorguille** [comment]
@dannon I think I'm at the edge of my limit about GitHub... How can I fork your project if I already have a fork of galaxy?

**Comment by bgruening** [comment]
@lecorguille you can do something like this:
```
git checkout -b dannon-i18n dev
git pull https://github.com/dannon/galaxy.git i18n
```

**Comment by dannon** [comment]
If you already have a fork, something like this:

```
git remote add dannon
git checkout -b i18n dannon/i18n
<make changes>
git push -u <your fork, probably 'origin'> i18n
```

And then issue a PR

**Comment by dannon** [comment]
(or what bjoern said!  too many ways to do things :))

**Comment by lecorguille** [comment]
Oh, finally, I was not at my limit. I'm just tried :/
I knew the answer
Ok, allons-y !

**Comment by yvanlebras** [comment]
Related to your sentence @lecorguille  
 
> Is it just for our famous bad skills in English? :P
> 
> Because as far as I know, I never met any French user stuck because they don't understand the user interface (They are because their bam are not sorted :/)

My new job until september is to challenge the use of Galaxy to give access to data and tools to citizen scientists.... and we need to have something written in French for some users (ie citizen/naturalists). I have an intern who will specifically challenge Galaxy on this point + other things related to GUI customization.... This PR is a superb opportunity to work deeper on that. If the "Galaxy challenge" become a success, it will be possible to have dedicated fundings to maintain this kind of improvement...

**Comment by yboursin** [comment]
Cool :)

**Comment by yvanlebras** [comment]
@lecorguille, you are working on the translation? If you want some help, don't hesitate ;)

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: +1 for "Visualisation"

**Comment by lecorguille** [comment]
@dannon before your merge, I have some minor changes.

**Comment by dannon** [comment]
Ok, no rush.  I'm going through and making sure it's all working well, too.

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: "Restaurer" -> "Afficher"

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: "Êtes-vous certains?" -> "Êtes-vous certains ?"

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: We should be agree before the merge about Dataset transalation.
If you really want "Jeu de données", can you sync the first part?

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: 'Get Data', I guess will not be translated.
So, IMHO, it's confusing!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: Proposition: "Le calcul ayant créé ce jeu de données" -> "Le processus à l'origine de ce jeu de données"

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js:248]: The same, we should choose for `job` between: Traitement, Calcul and Processus
I like Traitement

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js]: "avant avant" -> "avant"

**Comment by dannon** [comment]
Ready for review I think?

There is a ton more we can localize, but this gets the masthead and a lot more of the history panel.  Still need to do more tooltips, etc., but those can be separate PRs.

Thanks a ton to @yvanlebras, @lecorguille, and @bebatut for the translations!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js:294]: Vu :)
J'avais pensé à une contrepèterie mais je suis cul

File: client/galaxy/scripts/nls/fr/locale.js
Comment (lecorguille) [inline client/galaxy/scripts/nls/fr/locale.js:251]: \"Resume Paused Jobs\" is not currently translated :/

**Review by lecorguille — CHANGES_REQUESTED** [review]

File: client/galaxy/scripts/nls/fr/locale.js
Comment (dannon) [reply client/galaxy/scripts/nls/fr/locale.js]: Yeah, this is a sticky one.  Let me see what I can do about "Get Data", since it's so high profile.  We certainly won't be able to do all tools, but maybe this section.

**Comment by dannon** [comment]
Looks like the translation for 'search datasets' is a bit long:
image

Is there a shorter way to say this?

**Comment by lecorguille** [comment]
Yvan and its commit integration :/
What I proposed: ["Rechercher des données"](https://github.com/galaxyproject/galaxy/pull/3762/commits/542a84afd5552b94bfdaff4ca8a6c0c368f1489a#diff-c1c5f5030bd2946beef3740bb9a4ccc4R79)

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: Yes, this is better!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: +1

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: I think "jeu de données" is a better translation for "dataset" but maybe this can be context dependent..

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: +1 to translate at least this section and thus modify here the "Get data" statement.

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js]: Really better! Thanks

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js:248]: Sure! Once again, this seems context dependent I think... So maybe we can't always translate with the same french Word. ...

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js:251]: Same as for "Get Data"... If "Resume Paused Jobs" will be translated, thus we will update this statement I think

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js:294]: Oeil de lynx ;)

File: client/galaxy/scripts/nls/fr/locale.js
Comment (nagoue) [reply client/galaxy/scripts/nls/fr/locale.js]: Hi, this should be : "Êtes-vous certain ?"

File: client/galaxy/scripts/nls/fr/locale.js
Comment (dannon) [reply client/galaxy/scripts/nls/fr/locale.js:251]: "Get Data" is more difficult and I think I'll wait until a second pass, but I'm adding "Resume Paused Jobs" and the rest of that menu now.

Comment (dannon) [reply client/galaxy/scripts/nls/fr/locale.js:251]: @lecorguille @yvanlebras I added more translation points in 54af1979676c817b00044e645383e638b86fdcfa though unfortunately not for "Get Data" yet -- that one's more complicated and I'll want to address it in a future PR.   If you guys want to fill in those, we can swap "Resume Paused Jobs", etc.

File: client/galaxy/scripts/nls/fr/locale.js
Comment (dannon) [reply client/galaxy/scripts/nls/fr/locale.js:294]: Just to confirm, this is fine as written, correct?  :)

**Comment by lecorguille** [comment]
@dannon and @yvanlebras 
I can propose something today about [Add more translation points for localization.](https://github.com/galaxyproject/galaxy/pull/3762/commits/54af1979676c817b00044e645383e638b86fdcfa)

**Comment by ValentinChCloud** [comment]
Hi everyone . I'm Yvan's trainee, may I help you with this translation job?  I'm a bit a newbie on github, if errors are made, tell me. I can start on @lecorguille  suggestion.

**Comment by dannon** [comment]
@ValentinChCloud I just saw your other PR, sorry!  I think this pass is pretty much done now, but we can identify the next places we'll want to improve.  Feel free to ping me on IRC (or Gitter) to talk about what you might be able to do next!

**Comment by ValentinChCloud** [comment]
@dannon no worries ! Moreover I pulled it on galaxyproject  the first time ...  Thanks ,I will do it .

**Comment by lecorguille** [comment]
What a success!
@ValentinChCloud The next one is for you :)

**Comment by martenson** [comment]
Thanks all!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (mvdbeek) [reply client/galaxy/scripts/nls/fr/locale.js:294]: Well, it translates more or less to `dumbass, the dataset is already deleted`. We could be a bit friendlier here ;),
`"Le jeu de données est déjà supprimé ;)"` would be fine.

Comment (lecorguille) [reply client/galaxy/scripts/nls/fr/locale.js:294]: Oups, I forgot to fix this!

Comment (dannon) [reply client/galaxy/scripts/nls/fr/locale.js:294]: No worries, all resolved now!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js:251]: Thanks!

File: client/galaxy/scripts/nls/fr/locale.js
Comment (yvanlebras) [reply client/galaxy/scripts/nls/fr/locale.js:294]: A ba bravo ;) first proposition was better I think but majority have voted !

**Pull request merged**
